semana14/projeto-f4bank/src/createAccount.ts

```diff
@@ -0,0 +1,134 @@
+import { readFile, writeFile } from 'fs'
+import * as moment from 'moment'
+moment.locale('pt-br')
+
+const databasePath = './database.json' 
+
+type account = {
+    name: string,
+    cpf: string,
+    birthDay: string,
+    balance: number
+    extract: expense[],
+}
+
+type expense = {
+    value: number,
+    date: string,
+    description: string,
+}
+
+type accountsData = {
+    clientsAccount: account[]
+}
+
+
+const newAccount: account = {
+    name: process.argv[2],
+    cpf: process.argv[3],
+    birthDay: moment(process.argv[4], 'DD/MM/YYYY').format('DD/MM/YYYY'),
+    balance: 0,
+    extract: [],
+}
+
+
+const getAllAccounts = async (): Promise<accountsData> => {
+    const data = await readDatabase
+    const allAccounts: accountsData = JSON.parse(data)
+    return allAccounts
+}
+
+
+const readDatabase: Promise<string> = new Promise((resolve, reject) => {
+    readFile(databasePath, (err: Error, data: Buffer) => {
+        if(err){
+            reject(err)
+            return
+        }
+        resolve(data.toString())
+    })
+})
+
+
+const createAccount = async (accountList: accountsData) => {
+    accountList.clientsAccount.push(newAccount);
+    const allAccounts = JSON.stringify(accountList);
+    
+    const putClientsData: Promise<void> = new Promise((resolve, reject) => {
+        writeFile(databasePath, allAccounts, (err) => {
+            if(err){
+                reject(err)
+                return
+            }
+            resolve(console.log(`
+            -------------------------
+            Conta criada com sucesso!
+            -------------------------`))
```

Comment:
Hehe, adorei a formatação inusitada da mensagem, mas cuidado, o que você tá fazendo aí é dando `resolve(undefined)` pq o `console.log` não retorna nada, talvez fosse mais apropriado passar só essa string (a mensagem) como parâmetro do `resolve` e dar o console.log no que for retornado, exemplo:

```js
// troquei as quebras de linha por \n que dá na mesma e polui menos o código ;)

resolve(`\n ------------------------- \n Conta criada com sucesso! \n -------------------------`);
```

```js
putClientsData.then(mensagem => console.log(mensagem))
```